### PR TITLE
Expose api gateway event mocks

### PR DESCRIPTION
**Pull request by adriencaccia**
- feat(serverless-helpers): add api gateway mocks
- refactor(serverless-contracts): use mock helpers from serverless-helpers
- test(serverless-helpers): add handler unit test

### CODE REVIEW

File: packages/serverless-helpers/src/mocks/apiGateway.ts [hunk 1]
```diff
@@ -38,7 +38,7 @@ export const getRequestContextMock = (
   ...args,
 });
 
-export const getRequestContextMockV2 = (
+export const getAPIGatewayEventRequestContextMockV2 = (
```
Comment (fargito) [inline packages/serverless-helpers/src/mocks/apiGateway.ts:41]:
```suggestion
export const getAPIGatewayV2EventRequestContextMock = (
```
Wouldn't it be a better naming?

Comment (adriencaccia) [reply packages/serverless-helpers/src/mocks/apiGateway.ts]: Yep, changed it

**Review by fargito — APPROVED** [review]

**Comment by fargito** [comment]
@adriencaccia do you think this PR could be merged? It would nicely integrate with #358

**Comment by fargito** [comment]
@adriencaccia I rebased it on main but it seems to have a problem with tests.

I was event thinking, wouldn't it be great to expose a serverless-contracts feature called `buildTestEvent` to bring even more type-safety to tests?

It could be used like:
```
// handler.test.ts

const { event, context, callback } = buildHandlerTestEvent(myContract)({
    // type-safe values from the contract...
});

const result = await main(event, context, callback);

expect(result.xxxx).toBe('yyyy') // whatever assertion on the result

```

Therefore we would remove the need to directly instantiate `getAPIGatewayV2EventRequestContextMock` or other mock builders. This function would directly populate the event context in the test event.

WDYT?

**Comment by adriencaccia** [comment]
> @adriencaccia do you think this PR could be merged? It would nicely integrate with #358

Yeah, I'll look into it this afternoon

**Review by adriencaccia — COMMENTED** [review]
LGTM 🎉 
I cannot approve the PR since I am the one that created it 😅
You can merge it

**Comment by fargito** [comment]
Great! Thanks

**Pull request merged**